docs: add Juno v0.7.3 details (#921)
starknet-io/starknet-docs: components/Starknet/modules/starknet_versions/pages/juno_versions.adoc

See the official https://github.com/NethermindEth/juno[Juno GitHub repository] for more details.

== https://github.com/NethermindEth/juno/releases/tag/v0.7.3[v0.7.3]

> 🎉 Exciting updates have landed in the latest Juno release! Now with support for upcoming Starknet v0.12.3, we've also supercharged RPC performance with a new global class cache, allowing for higher request throughput and optimized resource usage. We've updated blockifier which includes an important wallet integration fix. On top of that, expect new metrics and ongoing enhancements to the P2P layer, among other improvements. Here’s what’s new:

Reviewer: ```suggestion
Support for upcoming Starknet v0.12.3, improved RPC performance with a new global class cache, allowing for higher request throughput and optimized resource usage. We've updated blockifier, which includes an important wallet integration fix. On top of that, expect new metrics and ongoing enhancements to the P2P layer, among other improvements. Here’s what’s new:
```
